README.md

```diff
@@ -56,172 +67,74 @@ FilterHeadersStatus ExampleContext::onRequestHeaders(uint32_t headers, bool end_
 ```
-
-### Ownership of the resulting .wasm files
-
-The compiled files may be owned by root.  To chown them add the follow lines to the Makefile and docker invocation:
-
+bazel build //:my_extension.wasm
 ```
-PROXY_WASM_CPP_SDK=/sdk
 
-all: myproject.wasm
-  chown ${uid}.${gid} $^
+After the command finishes, you should be able to find `my_extension.wasm` file under `./bazel-bin` folder.
 
-include ${PROXY_WASM_CPP_SDK}/Makefile.base_lite
-```
-
-Invocation file (e.g. build.sh):
-
-```bash
-#!/bin/bash
-docker run -e uid="$(id -u)" -e gid="$(id -g)" -v $PWD:/work -w /work wasmsdk:v2 /build_wasm.sh
-```
-
-## Dependencies for building WASM modules:
-
-If you do not wish to use the Docker file, the dependencies can be installed by script (sdk\_container.sh), or by hand.
-
-### protobuf v3.9.1
-
-You must install the version of protobuf on your build system that matches the libprotobuf.a files (without any patches) so that the generated code matches the .a library.  Currently this is based on tag v3.9.1 of https://github.com/protocolbuffers/protobuf.
-
-```bash
-git clone https://github.com/protocolbuffers/protobuf
-cd protobuf
-git checkout v3.9.1
-git submodule update --init --recursive
-./autogen.sh
-./configure
-make
-make check
-sudo make install
-```
-
-### emscripten
-
-```bash
-git clone https://github.com/emscripten-core/emsdk.git
-cd emsdk
-./emsdk update-tags
-./emsdk install 2.0.7
-./emsdk activate 2.0.7
+### Compiling with the Docker build image
 
-source ./emsdk\_env.sh
-```
+A Dockerfile for the C++ SDK is provided in Dockerfile-sdk.
 
-It is possible later versions will work, e.g.
+It can built in this directory by:
 
-```bash
-./emsdk update-tags
-./emsdk install latest
-./emsdk activate latest
 ```
-
-However 2.0.7 is known to work.
-
-### Rebuilding the libprotobuf.a files
-
-If want to rebuild the libprotobuf.a files or use a different version see the instructions at https://github.com/kwonoj/protobuf-wasm. Commit 4bba8b2f38b5004f87489642b6ca4525ae72fe7f works for protobuf v3.9.x.
-
-```bash
-git clone https://github.com/protocolbuffers/protobuf protobuf-wasm
-cd protobuf-wasm
-git checkout v3.9.1
-git clone https://github.com/kwonoj/protobuf-wasm wasm-patches
-cd wasm-patches && git checkout 4bba8b2f38b5004f87489642b6ca4525ae72fe7f && cd ..
-git apply wasm-patches/*.patch
-./autogen.sh
-emconfigure ./configure --disable-shared CXXFLAGS="-O3 -flto"
-emmake make
-cd ..
-cp protobuf-wasm/src/.libs/libprotobuf-lite.a ${CPP_API}/libprotobuf-lite.a
-cp protobuf-wasm/src/.libs/libprotobuf.a ${CPP_API}/libprotobuf.a
+docker build -t wasmsdk -f Dockerfile-sdk .
 ```
 
-### WAVM binaries
+The docker image can be used for compiling wasm files:
 
 ```bash
-git clone git@github.com:WAVM/WAVM.git
-cd WAVM
-cmake "."
-make
-sudo make install
-```
-
-Note: ensure /usr/local/bin is in your path
-
+docker run \
+-u $(id -u):$(id -g) \
+-e HOME="/home" \
```

Comment:
This should probably be `${HOME}`, but do we actually need it?

Reply:
bazel needs `${HOME}` to find the place to create its cache, otherwise it will just create the cache under `/`. For simplicity I use `/home`, otherwise we'd need to expand the runtime command to `mkdir ${HOME}` in the container which seems to be unnecessary.